contracts/KNS_Retirer_v1.sol

```diff
@@ -225,20 +194,20 @@ contract KNS_Retirer is Initializable, OwnableUpgradeable {
         string memory _domainName,
         string memory _retMessage
     ) private {
-        BCTFromSwap = _swapUSDCToBCT(_USDCAmt);
+        uint BCTFromSwap = _swapUSDCToBCT(_USDCAmt);
         IRetire KI_Retirer = IRetire(Aggregator);
         IERC20(BCT).approve(Aggregator, BCTFromSwap);
         KI_Retirer.retireCarbon(
-            BCT,
-            BCT,
-            BCTFromSwap,
-            false,
+            BCT, // _sourceToken
+            BCT, // _poolToken
+            BCTFromSwap, // _amount
+            true, // _amountInCarbon
```

Comment:
@rajasmasala This was set to `false` before, but it should now be `true` because BCT is being sent, not USDC (so that the aggregator contract knows it doesn't have to swap it). Please correct me if I misunderstood that.